### ISSUE TITLE
[apidoc] '.' as <module_path> cause an error

### ISSUE THREAD

**Issue by BerlinNurse** [sphinx-doc/sphinx#1823]
I am using Sphinx v1.3.1 (for Python3, insalled with pip3) on an Lubuntu 14.04.2 using Python 3.4.

I run sphinx-apidoc directly in the root of my source directory. So I really don't need to specifiy a <modul_path>. I just say '.' to let apidoc know that the current workind dir is the <modul_path>.

The call with the error looks like this

``` bash
user@MONSTER:~/share/work/MuckiBude$ sphinx-apidoc -F -o doc/src .
Creating file doc/src/MuckiBude.rst.
Traceback (most recent call last):
  File "/usr/lib/python3.4/encodings/idna.py", line 165, in encode
    raise UnicodeError("label empty or too long")
UnicodeError: label empty or too long

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/local/bin/sphinx-apidoc", line 11, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.4/dist-packages/sphinx/apidoc.py", line
374, in main qs.generate(d, silent=True, overwrite=opts.force)
  File "/usr/local/lib/python3.4/dist-packages/sphinx/quickstart.py",
line 1303, in generate d['project_url'] =
urlquote(d['project'].encode('idna')) UnicodeError: encoding with
'idna' codec failed (UnicodeError: label empty or too long)
```

I checked with debug code that "label" is empty and the given "input" to the error-raising function is the point.

When specifing the modul_path as an absolut path there is no problem.

**Comment by shimizukawa** [comment]
For now, sphinx-apidoc expects letters as a module_path instead of `.`
However error traceback you reported is not difficult to understand.
I'll fix this behavior.

FYI. sphinx-apidoc `-H` option also avoid this problem.

**Comment by shimizukawa** [comment]
Now '.' is converted to absolute path automatically.

**Comment by InterestsFantastic** [comment]
I just ran into the same error.

**Comment by InterestsFantastic** [comment]
I just ran into the same error.

**Comment by InterestsFantastic** [comment]
I think the issue is that there are users (like me) who want to just do something like:

import sphinx
sphinx.make_autodoc()

and have it default to spitting out to the cwd with .rst that displays well on github.

We expect press button, receive bacon, and to be honest, that's desirable, as there are a lot of noobs to programming that want to get started on github and want to make nice looking projects that people can collaborate on.

**Comment by tk0miya** [comment]
@DavidCaughell This issue was closed 2+ years ago. So I guess your error might be different with original issue. To investigate your report, I'd like to know your situation.
Could you file a new issue please?

**Comment by InterestsFantastic** [comment]
Sorry but I've already uninstalled. I'm trying to write a little script to do as I described earlier. When I come back to this in a more professional way (i.e. needing to make good docs with a good program) I will likely use sphinx and definitely report bugs if I do. I got to this page by googling the error message. I think if you try to use '.' as the argument for module path sphinx doesn't like that.
